src/field/FieldStepper.hh

```diff
@@ -0,0 +1,73 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldStepper.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "FieldEquation.hh"
+
+#include "field/base/OdeArray.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Brief class description.
+ *
+ * This is the base class of field steppers
+ */
+template<typename T>
+class FieldStepper
```

Comment:
What's the template parameter here? Also the class documentation (and other places where "Brief class description." appear) need to be updated.

Reply:
Added "\tparam T A derived class.".  Will add more descriptions for other classes.

Reply:
I still don't understand what that means here -- is this a "curiously recursive template pattern"? What interface is T expected to have?

Reply:
Yes, it is CRTP.